src/profile-logic/sanitize.js

```diff
@@ -62,17 +62,47 @@ export function sanitizePII(
   const PIIToBeRemoved = maybePIIToBeRemoved;
   const oldThreadIndexToNew: Map<ThreadIndex, ThreadIndex> = new Map();
 
-  const windowIdToBeSanitized = new Set();
+  // This set keeps the ids to be removed when removing private browsing data.
+  const windowIdFromPrivateBrowsing = new Set();
+  // This set keeps the id that are from the active tab id, when we want to
+  // remove the data of other tabs.
+  // Note that in the case of an id is both from private browsing and from
+  // the active tab, it will be present in both.
+  const windowIdFromActiveTab = new Set();
+
   let pages = profile.pages;
   if (pages) {
-    if (PIIToBeRemoved.shouldRemovePrivateBrowsingData) {
+    if (
+      PIIToBeRemoved.shouldRemovePrivateBrowsingData ||
+      PIIToBeRemoved.shouldKeepTabID !== null
+    ) {
       // slicing here so that we can mutate it later.
       pages = pages.slice();
 
       for (let i = pages.length - 1; i >= 0; i--) {
         const page = pages[i];
-        if (page.isPrivateBrowsing) {
-          windowIdToBeSanitized.add(page.innerWindowID);
+
+        let removePageEntry = false;
```

Comment:
We need to make sure we don't "splice" twice, that's the reason for this variable.